### PR TITLE
Add instructions for using IoT credentials in the samples and document additional cmake options

### PR DESCRIPTION
**Pull request by sirknightj**
*What was changed?*
The readme was changed to include better instructions.

*Why was it changed?*
Because the information was missing from the readme.

*How was it changed?*
The instructions were added to the readme.

*What testing was done for the changes?*
I followed the instructions on my Mac and on an Ubuntu instance.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,7 +154,7 @@ You can pass the following options to `cmake ..`.
 * `-DCMAKE_BUILD_TYPE` -- Build Release/Debug libraries. By default, the SDK generates Release build. The standard options are listed [here](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#default-and-custom-configurations)
 * `-DLINK_PROFILER` -- Link with gperftools (available profiler options are listed [here](https://github.com/gperftools/gperftools))
 * `-DPKG_CONFIG_EXECUTABLE` -- Set pkg config path. This might be required to find gstreamer's pkg config specifically on Windows.
-* `DENABLE_KVS_THREADPOOL` -- Enable the KVS threadpool which is off by default.
+* `-DENABLE_KVS_THREADPOOL` -- Enable the KVS threadpool which is off by default.
```
Comment (stefankiesz) [inline README.md:157]: Whoops

**Review by stefankiesz — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -402,6 +410,27 @@ createLwsIotCredentialProvider(
...
+
+```shell
+export AWS_IOT_CORE_CREDENTIAL_ENDPOINT=xxxxx.credentials.iot.xxxxx.amazonaws.com
+export AWS_IOT_CORE_PRIVATE_KEY=xxxxxxxx-private.pem.key 
+export AWS_IOT_CORE_ROLE_ALIAS=xxxxxx
+export AWS_IOT_CORE_THING_NAME=xxxxxx
+export AWS_IOT_CORE_CERT=xxxxx-certificate.pem.crt
+```
+
+AWS access keys are ignored if the sample was built in IoT credentials mode.
```
Comment (unicornss) [inline README.md:432]: AK/SK/ST set in env?

**Review by unicornss — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (sirknightj) [reply README.md:432]: Correct.

https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/8248241cec28b651839d91f77a706f4fd028dbeb/samples/Common.c#L759-L769

File: README.md [hunk 3]
```diff
@@ -402,6 +410,27 @@ createLwsIotCredentialProvider(
 freeIotCredentialProvider(&pSampleConfiguration->pCredentialProvider);
 ```
 
+### Running samples with IoT
```
[the remark this answers is not in the document]
Comment (sirknightj) [reply README.md]: Adjusted

**Pull request merged**